fastlane/Fastfile

```diff
@@ -172,16 +176,49 @@ platform :ios do
   #####################################################################################
   desc 'Creates a new release branch from the current trunk'
   lane :code_freeze do |options|
-    ios_codefreeze_prechecks(options)
+    # Verify that there's nothing in progress in the working copy
+    ensure_git_status_clean
+
+    # Check out the up-to-date default branch, the designated starting point for the code freeze
+    Fastlane::Helper::GitHelper.checkout_and_pull(DEFAULT_BRANCH)
+
+    message = <<-MESSAGE
+
+      Code Freeze:
+      • New release branch from #{DEFAULT_BRANCH}: release/#{release_version_next}
+      • Current release version and build code: #{release_version_current} (#{build_code_current}).
+      • New release version and build code: #{release_version_next} (#{build_code_code_freeze}).
+
+    MESSAGE
+
+    UI.important(message)
+    UI.user_error!('Aborted by user request') if !options[:skip_confirm] && !UI.confirm('Do you want to continue?')
+
+    # Create the release branch
+    UI.message('Creating release branch...')
+    ensure_git_branch(branch: DEFAULT_BRANCH)
+    Fastlane::Helper::GitHelper.create_branch("release/#{release_version_next}", from: DEFAULT_BRANCH)
+    UI.success("Done! New release branch is: #{git_branch}")
 
-    ios_bump_version_release
-    new_version = app_version
+    # Bump the release version and build code and write it to the `xcconfig` file
+    UI.message('Bumping release version and build code...')
+    VERSION_FILE.write(
+      version_short: release_version_next,
+      version_long: build_code_code_freeze
+    )
+    Fastlane::Helper::Ios::GitHelper.commit_version_bump
```

Comment:
I run into an issue with this during the 23.6 Jetpack/WordPress code freeze and ended up replacing it with `git_commit` in https://github.com/wordpress-mobile/WordPress-iOS/pull/21911/commits/f886436c4fcdbaf6d4fd3a1f65e06a3fc06b0b8e

More info in https://github.com/wordpress-mobile/release-toolkit/issues/525

How would you feel about doing the same here? Otherwise, because of the `ENV['PROJECT_ROOT_FOLDER']` above, I fear this will break on the next code freeze.

Reply:
Shoot, I missed that method during the env var project because it referenced the env var like `get_from_env!(key: 'PROJECT_ROOT_FOLDER')` instead of `ENV['PROJECT_ROOT_FOLDER']`. Sorry, I should have caught that. I'll make sure to get the other iOS apps updated. 

Fixed for this PR in https://github.com/woocommerce/woocommerce-ios/commit/97204c6a0430e90c290e9d72147eb9d993b8af4d